In [42]:
# IMPORTING DEPENDENCIES :-
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import OneHotEncoder, StandardScaler,LabelEncoder
from sklearn.ensemble import RandomForestRegressor
from sklearn.metrics import mean_squared_error, r2_score
from sklearn.metrics import accuracy_score, precision_score, recall_score, f1_score
from sklearn.metrics import confusion_matrix,classification_report
import re

In [11]:
# reading the dataset :- 
dataset = pd.read_csv('/content/drive/MyDrive/datasets/Data_Science_Internship - Dump.csv')
dataset.head()

Unnamed: 0                                           Agent_id status  \
0           0  1deba9e96f404694373de9749ddd1ca8aa7bb823145a6f...   LOST   
1           1  299ae77a4ef350ae0dd37d6bba1c002d03444fb1edb236...   LOST   
2           2  c213697430c006013012dd2aca82dd9732aa0a1a6bca13...   LOST   
3           3  eac9815a500f908736d303e23aa227f0957177b0e6756b...   LOST   
4           4  1deba9e96f404694373de9749ddd1ca8aa7bb823145a6f...   LOST   

      lost_reason                budget                                lease  \
0  Not responding                   NaN                                  NaN   
1      Low budget                   NaN                                  NaN   
2  Not responding  £121 - £180 Per Week  Full Year Course Stay 40 - 44 weeks   
3      Low budget                   0-0                                    0   
4       Junk lead                   NaN                                  NaN   

     movein                                             source  \
0       NaN  9b2d5b4678781e53038e91ea5324530a03f27dc1d0e5f6...   
1       NaN  9b2d5b4678781e53038e91ea5324530a03f27dc1d0e5f6...   
2  31/08/22  7aae3e886e89fc1187a5c47d6cea1c22998ee610ade1f2...   
3       NaN  ba2d0a29556ac20f86f45e4543c0825428cba33fd7a9ea...   
4       NaN  9b2d5b4678781e53038e91ea5324530a03f27dc1d0e5f6...   

                                         source_city  \
0  9b2d5b4678781e53038e91ea5324530a03f27dc1d0e5f6...   
1  9b2d5b4678781e53038e91ea5324530a03f27dc1d0e5f6...   
2  9b8cc3c63cdf447e463c11544924bf027945cbd29675f7...   
3  a5f0d2d08eb0592087e3a3a2f9c1ba2c67cc30f2efd2bd...   
4  9b2d5b4678781e53038e91ea5324530a03f27dc1d0e5f6...   

                                      source_country  \
0  9b2d5b4678781e53038e91ea5324530a03f27dc1d0e5f6...   
1  9b2d5b4678781e53038e91ea5324530a03f27dc1d0e5f6...   
2  e09e10e67812e9d236ad900e5d46b4308fc62f5d69446a...   
3  e09e10e67812e9d236ad900e5d46b4308fc62f5d69446a...   
4  9b2d5b4678781e53038e91ea5324530a03f27dc1d0e5f6...   

                                          utm_source  \
0  3d59f7548e1af2151b64135003ce63c0a484c26b9b8b16...   
1  3d59f7548e1af2151b64135003ce63c0a484c26b9b8b16...   
2  bbdefa2950f49882f295b1285d4fa9dec45fc4144bfb07...   
3  bbdefa2950f49882f295b1285d4fa9dec45fc4144bfb07...   
4  3d59f7548e1af2151b64135003ce63c0a484c26b9b8b16...   

                                          utm_medium  \
0  268ad70eb5bc4737a2ae28162cbca30118cc94520e49ef...   
1  268ad70eb5bc4737a2ae28162cbca30118cc94520e49ef...   
2  09076eb7665d1fb9389c7c4517fee0b00e43092eb34821...   
3  09076eb7665d1fb9389c7c4517fee0b00e43092eb34821...   
4  268ad70eb5bc4737a2ae28162cbca30118cc94520e49ef...   

                                            des_city  \
0  ecc0e7dc084f141b29479058967d0bc07dee25d9690a98...   
1  5372372f3bf5896820cb2819300c3e681820d82c6efc54...   
2  11ab03a1a8c367191355c152f39fe28cae5e426fce49ef...   
3  19372fa44c57a01c37a5a8418779ca3d99b0b59731fb35...   
4  9b2d5b4678781e53038e91ea5324530a03f27dc1d0e5f6...   

                                         des_country room_type  \
0  8d23a6e37e0a6431a8f1b43a91026dcff51170a89a6512...       NaN   
1  8d23a6e37e0a6431a8f1b43a91026dcff51170a89a6512...       NaN   
2  8d23a6e37e0a6431a8f1b43a91026dcff51170a89a6512...   Ensuite   
3  8d23a6e37e0a6431a8f1b43a91026dcff51170a89a6512...       NaN   
4  9b2d5b4678781e53038e91ea5324530a03f27dc1d0e5f6...       NaN   

                                             lead_id  
0  cd5dc0d9393f3980d11d4ba6f88f8110c2b7a7f7796307...  
1  b94693673a5f7178d1b114e4004ad52377d3244dd24a3d...  
2  96ea4e2bf04496c044745938c0299c264c3f4ba079e572...  
3  1d2b34d8add02a182a4129023766ca4585a8ddced0e5b3...  
4  fc10fffd29cfbe93c55158fb47752a7501c211d253468c...

In [12]:

dataset#  removing all  the rows except label "WON","LOST"['status'].count() 

Unnamed: 0                                           Agent_id status  \
0               0  1deba9e96f404694373de9749ddd1ca8aa7bb823145a6f...   LOST   
1               1  299ae77a4ef350ae0dd37d6bba1c002d03444fb1edb236...   LOST   
2               2  c213697430c006013012dd2aca82dd9732aa0a1a6bca13...   LOST   
3               3  eac9815a500f908736d303e23aa227f0957177b0e6756b...   LOST   
4               4  1deba9e96f404694373de9749ddd1ca8aa7bb823145a6f...   LOST   
...           ...                                                ...    ...   
46603       46603  2306878a9ad9b57686cd623dd285aaa9b25afdf627f651...   LOST   
46604       46604  327ec29056cc47c24bf922f7dc0f78261dad5c726d7353...   LOST   
46605       46605  1134c0a7d44fdae1afd7f1f64e2789496784095ca0a050...   LOST   
46606       46606  8b8b029f1142f5cbc825aa6cbee01406c915c6b055db79...   LOST   
46607       46607  1ea65ea38f2f574b3875ba895e4ff76b284b7725041612...   LOST   

            lost_reason                budget  \
0        Not responding                   NaN   
1            Low budget                   NaN   
2        Not responding  £121 - £180 Per Week   
3            Low budget                   0-0   
4             Junk lead                   NaN   
...                 ...                   ...   
46603  Low availability   £60 - £120 Per week   
46604     Semester stay   £60 - £120 Per week   
46605  Low availability  £241 - £300 Per week   
46606  Low availability                  1108   
46607  Low availability  £181 - £240 Per Week   

                                            lease    movein  \
0                                             NaN       NaN   
1                                             NaN       NaN   
2             Full Year Course Stay 40 - 44 weeks  31/08/22   
3                                               0       NaN   
4                                             NaN       NaN   
...                                           ...       ...   
46603  Complete Education Year Stay 50 - 52 weeks  01/09/22   
46604              Summer/Short Stay 8 - 12 weeks  29/09/22   
46605         Full Year Course Stay 40 - 44 weeks  20/09/22   
46606                                         294  30/08/22   
46607         Full Year Course Stay 40 - 44 weeks  01/09/22   

                                                  source  \
0      9b2d5b4678781e53038e91ea5324530a03f27dc1d0e5f6...   
1      9b2d5b4678781e53038e91ea5324530a03f27dc1d0e5f6...   
2      7aae3e886e89fc1187a5c47d6cea1c22998ee610ade1f2...   
3      ba2d0a29556ac20f86f45e4543c0825428cba33fd7a9ea...   
4      9b2d5b4678781e53038e91ea5324530a03f27dc1d0e5f6...   
...                                                  ...   
46603  9b2d5b4678781e53038e91ea5324530a03f27dc1d0e5f6...   
46604  9b2d5b4678781e53038e91ea5324530a03f27dc1d0e5f6...   
46605  7aae3e886e89fc1187a5c47d6cea1c22998ee610ade1f2...   
46606  d684761c17c11590f6e2525b48141cb2c0c6f2be5df4e2...   
46607  7aae3e886e89fc1187a5c47d6cea1c22998ee610ade1f2...   

                                             source_city  \
0      9b2d5b4678781e53038e91ea5324530a03f27dc1d0e5f6...   
1      9b2d5b4678781e53038e91ea5324530a03f27dc1d0e5f6...   
2      9b8cc3c63cdf447e463c11544924bf027945cbd29675f7...   
3      a5f0d2d08eb0592087e3a3a2f9c1ba2c67cc30f2efd2bd...   
4      9b2d5b4678781e53038e91ea5324530a03f27dc1d0e5f6...   
...                                                  ...   
46603  7d1550b844ff586a6023216c06263105eed0a849a2a1f6...   
46604  9f2ba6981e36ea0fca6c16f42e6413d788d2a7499b8b7f...   
46605  7d1550b844ff586a6023216c06263105eed0a849a2a1f6...   
46606  e9c722cbefc2f055ae60b4e2cbe73a2d99537eab0c37f3...   
46607  1496126b7015defbc3b1d129a46096a5c135075aff2add...   

                                          source_country  \
0      9b2d5b4678781e53038e91ea5324530a03f27dc1d0e5f6...   
1      9b2d5b4678781e53038e91ea5324530a03f27dc1d0e5f6...   
2      e09e10e67812e9d236ad900e5d46b4308fc62f5d69446a...   
3      e09e10e67812e9d236ad900

DATA CLEANING
 

In [6]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [13]:
# REMOVING ROWS WITH STATUS OTHER THAN 'WON' OR 'LOSE'

In [22]:
dataset = dataset[dataset['status'].isin(['WON','LOST'])]

In [23]:
# total number of elements in status column after only considering 'won' and 'lose' categories :- 
dataset['status'].count()

46317

In [24]:
# checking total number of null values present in our dataset  :-
print(dataset.isnull().sum())

Unnamed: 0            0
Agent_id              0
status                0
lost_reason        3073
budget             3694
lease              2336
movein            13610
source             5951
source_city        8831
source_country     8622
utm_source           61
utm_medium         3184
des_city           2529
des_country        2529
room_type         23491
lead_id               0
dtype: int64


In [25]:
#REPLACING '9b2d5b4678781e53038e91ea5324530a03f27dc1d0e5f6c9bc9d493a23be9de0' to  NAN VALUE:--
dataset = dataset.replace('9b2d5b4678781e53038e91ea5324530a03f27dc1d0e5f6c9bc9d493a23be9de0', np.nan)

In [26]:
# total number of null values presen-
dataset.isnull().sum()

Unnamed: 0            0
Agent_id              0
status                0
lost_reason        3073
budget             3694
lease              2336
movein            13610
source             5951
source_city        8831
source_country     8622
utm_source           61
utm_medium         3184
des_city           2529
des_country        2529
room_type         23491
lead_id               0
dtype: int64

In [27]:
#  considering only data which has <10% none values
cols = []

for i in dataset.columns:
  if dataset[i].isnull().sum() < len(data)/10  :
    cols.append(i)

In [28]:
cols

['Unnamed: 0',
 'Agent_id',
 'status',
 'lost_reason',
 'budget',
 'lease',
 'utm_source',
 'utm_medium',
 'des_city',
 'des_country',
 'lead_id']

In [29]:
# creating a dataframe of the above taken columns :-
df = dataset.loc[:,cols]

In [30]:
df.head()

Unnamed: 0                                           Agent_id status  \
0           0  1deba9e96f404694373de9749ddd1ca8aa7bb823145a6f...   LOST   
1           1  299ae77a4ef350ae0dd37d6bba1c002d03444fb1edb236...   LOST   
2           2  c213697430c006013012dd2aca82dd9732aa0a1a6bca13...   LOST   
3           3  eac9815a500f908736d303e23aa227f0957177b0e6756b...   LOST   
4           4  1deba9e96f404694373de9749ddd1ca8aa7bb823145a6f...   LOST   

      lost_reason                budget                                lease  \
0  Not responding                   NaN                                  NaN   
1      Low budget                   NaN                                  NaN   
2  Not responding  £121 - £180 Per Week  Full Year Course Stay 40 - 44 weeks   
3      Low budget                   0-0                                    0   
4       Junk lead                   NaN                                  NaN   

                                          utm_source  \
0  3d59f7548e1af2151b64135003ce63c0a484c26b9b8b16...   
1  3d59f7548e1af2151b64135003ce63c0a484c26b9b8b16...   
2  bbdefa2950f49882f295b1285d4fa9dec45fc4144bfb07...   
3  bbdefa2950f49882f295b1285d4fa9dec45fc4144bfb07...   
4  3d59f7548e1af2151b64135003ce63c0a484c26b9b8b16...   

                                          utm_medium  \
0  268ad70eb5bc4737a2ae28162cbca30118cc94520e49ef...   
1  268ad70eb5bc4737a2ae28162cbca30118cc94520e49ef...   
2  09076eb7665d1fb9389c7c4517fee0b00e43092eb34821...   
3  09076eb7665d1fb9389c7c4517fee0b00e43092eb34821...   
4  268ad70eb5bc4737a2ae28162cbca30118cc94520e49ef...   

                                            des_city  \
0  ecc0e7dc084f141b29479058967d0bc07dee25d9690a98...   
1  5372372f3bf5896820cb2819300c3e681820d82c6efc54...   
2  11ab03a1a8c367191355c152f39fe28cae5e426fce49ef...   
3  19372fa44c57a01c37a5a8418779ca3d99b0b59731fb35...   
4                                                NaN   

                                         des_country  \
0  8d23a6e37e0a6431a8f1b43a91026dcff51170a89a6512...   
1  8d23a6e37e0a6431a8f1b43a91026dcff51170a89a6512...   
2  8d23a6e37e0a6431a8f1b43a91026dcff51170a89a6512...   
3  8d23a6e37e0a6431a8f1b43a91026dcff51170a89a6512...   
4                                                NaN   

                                             lead_id  
0  cd5dc0d9393f3980d11d4ba6f88f8110c2b7a7f7796307...  
1  b94693673a5f7178d1b114e4004ad52377d3244dd24a3d...  
2  96ea4e2bf04496c044745938c0299c264c3f4ba079e572...  
3  1d2b34d8add02a182a4129023766ca4585a8ddced0e5b3...  
4  fc10fffd29cfbe93c55158fb47752a7501c211d253468c...

In [31]:
# here we are replacing the nan values in the columns with the most frequent values of the particular column :-
data_clean = df.apply(lambda x: x.fillna(x.value_counts().index[0]))

In [32]:
# dropping 'Unnamed: 0','lead_id' as they have no effect on our label
data_clean = data_clean.drop(['Unnamed: 0'],axis = 1)
data_clean.drop(labels=data_clean.columns[-1],axis=1,inplace=True)
data_clean.drop(labels=data_clean.columns[0],axis=1,inplace=True)


In [33]:
# remaining features 
data_clean.columns

Index(['status', 'lost_reason', 'budget', 'lease', 'utm_source', 'utm_medium',
       'des_city', 'des_country'],
      dtype='object')

In [34]:
data_clean.head()

status     lost_reason                budget  \
0   LOST  Not responding   £60 - £120 Per week   
1   LOST      Low budget   £60 - £120 Per week   
2   LOST  Not responding  £121 - £180 Per Week   
3   LOST      Low budget                   0-0   
4   LOST       Junk lead   £60 - £120 Per week   

                                 lease  \
0  Full Year Course Stay 40 - 44 weeks   
1  Full Year Course Stay 40 - 44 weeks   
2  Full Year Course Stay 40 - 44 weeks   
3                                    0   
4  Full Year Course Stay 40 - 44 weeks   

                                          utm_source  \
0  3d59f7548e1af2151b64135003ce63c0a484c26b9b8b16...   
1  3d59f7548e1af2151b64135003ce63c0a484c26b9b8b16...   
2  bbdefa2950f49882f295b1285d4fa9dec45fc4144bfb07...   
3  bbdefa2950f49882f295b1285d4fa9dec45fc4144bfb07...   
4  3d59f7548e1af2151b64135003ce63c0a484c26b9b8b16...   

                                          utm_medium  \
0  268ad70eb5bc4737a2ae28162cbca30118cc94520e49ef...   
1  268ad70eb5bc4737a2ae28162cbca30118cc94520e49ef...   
2  09076eb7665d1fb9389c7c4517fee0b00e43092eb34821...   
3  09076eb7665d1fb9389c7c4517fee0b00e43092eb34821...   
4  268ad70eb5bc4737a2ae28162cbca30118cc94520e49ef...   

                                            des_city  \
0  ecc0e7dc084f141b29479058967d0bc07dee25d9690a98...   
1  5372372f3bf5896820cb2819300c3e681820d82c6efc54...   
2  11ab03a1a8c367191355c152f39fe28cae5e426fce49ef...   
3  19372fa44c57a01c37a5a8418779ca3d99b0b59731fb35...   
4  ecc0e7dc084f141b29479058967d0bc07dee25d9690a98...   

                                         des_country  
0  8d23a6e37e0a6431a8f1b43a91026dcff51170a89a6512...  
1  8d23a6e37e0a6431a8f1b43a91026dcff51170a89a6512...  
2  8d23a6e37e0a6431a8f1b43a91026dcff51170a89a6512...  
3  8d23a6e37e0a6431a8f1b43a91026dcff51170a89a6512...  
4  8d23a6e37e0a6431a8f1b43a91026dcff51170a89a6512...

In [36]:
# All values are converted to nonnull 
data_clean.isna().sum()

status         0
lost_reason    0
budget         0
lease          0
utm_source     0
utm_medium     0
des_city       0
des_country    0
dtype: int64

In [ ]:
# finding the number of unique values of the column :-
for col in data_clean.columns:

  print(col ," = ",len(data_clean[col].unique()))

status  =  2
lost_reason  =  30
budget  =  1851
lease  =  311
utm_source  =  34
utm_medium  =  63
des_city  =  219
des_country  =  14


In [ ]:
# Unique catagorical values of the lease
data_clean['lease'].unique()

array(['Full Year Course Stay 40 - 44 weeks', '0', 'semester-stay', '51',
       'Semester Stay 20 - 24 weeks',
       'Complete Education Year Stay 50 - 52 weeks', 'full-course-stay',
       'full-year-stay', 'Other', '51 weeks', '44', '18',
       'full year stay', '51weeks', 'Summer/Short Stay 8 - 12 weeks',
       '51 weeks \\', 'other', 'short-stay', '52 weeks', '44 weeks', '12',
       '1 year', '96', '43', '20', '294', '20 - 24 weeks', '1 year .',
       '48', '40', 'N51', '6', 'Academic year', '8', '51Weeks',
       '50 weeks', '84', '15 weeks', 'full course', '45', '39', '1',
       '5 weeks', '50', '12-24', '25', '45 weeks', '19', '44-51',
       '6 weeks', 'FULL YEAR', '24-36', '32', '44/week', '42', '3 month',
       '44-51 weeks', '3 months', '24', '6 months', '2 years', '10', '52',
       '2 weeks', '46', 'sem stay', '13', 'lease', 'short stay', '4',
       '52weeks', '69', '16', '50 - 52 weeks', '44 - 51 weeks',
       '4-8 weeks', '4 weeks', '3', 'Full year stay', '11',

as we can see that the unique values in lease column are same to some values, for ex. 2 months, 2 month are same, so we are now applying regex to correct the spelling and finding a pattern in it so that we can reduce the unique values

In [43]:
#CODE TO CORRECT THE SPELLING MISTAKES OF THE CATEGORICAL COLUMNS :-
import re

def correct_spelling(text):
    # replace "sem" or "semester" with "semester"
    text = re.sub(r'\b(sem(?:ester)?)\b', 'semester', text, flags=re.IGNORECASE)
    # replace "stay" or "stya" with "stay"
    text = re.sub(r'\bstay(?:a)?\b', 'stay', text, flags=re.IGNORECASE)
    # replace "full" with "full"
    text = re.sub(r'\b(full)\b', 'full', text, flags=re.IGNORECASE)
    # replace "year" or "yyear" with "year"
    text = re.sub(r'\b(y(?:ea)?r)\b', 'year', text, flags=re.IGNORECASE)
    # replace "short" or "shotr" with "short"
    text = re.sub(r'\b(sh(?:or)?t)\b', 'short', text, flags=re.IGNORECASE)
    # replace "week" or "weeeks" with "week"
    text = re.sub(r'\b(week(?:s)?)\b', 'week', text, flags=re.IGNORECASE)
    text = re.sub(r'\b(\d+\s*weeks?)\b', 'week', text, flags=re.IGNORECASE)
    # replace "month" or "motnhs" with "month"
    text = re.sub(r'\b(month(?:s)?)\b', 'month', text, flags=re.IGNORECASE)
    # replace "long" or "tearm" with "long term"
    text = re.sub(r'\blong\b|\btearm\b', 'long term', text, flags=re.IGNORECASE)
    # replace "academic" with "academic"
    text = re.sub(r'\b(academic)\b', 'academic', text, flags=re.IGNORECASE)
    return text


In [44]:
# correcting the spellings of the lease values :-
data_clean['lease'] = data_clean['lease'].apply(correct_spelling)

Still the values can be reduced furthure

In [45]:
# FUNCTION TO REDUCE THE UNIQUE VALUES BY THE HELP OF REGULAR EXPRESSION :-
def clean_lease(lease_value):

  if re.search(pattern="^[0-9 ]*-[0-9 ]*(weeks|WEEKS|Weeks|Week|week)$",string=lease_value):

    return lease_value[0:re.search("(weeks|WEEKS|Weeks|Week|week)$",string=lease_value).start()]

  elif re.search(pattern="^[0-9 ]*-[0-9 ]*(weeks|WEEKS|Weeks|Week|week)*$",string=lease_value):

    return lease_value[re.search("[0-9 ]*",string=lease_value).start():re.search("^[0-9 ]*",string=lease_value).end()]

  elif re.search(pattern="^[0-9 ]*(weeks|WEEKS|Weeks|Week|week)*$",string=lease_value):

    return lease_value[re.search("[0-9 ]*",string=lease_value).start():re.search("(weeks|WEEKS|Weeks|Week|week)*",string=lease_value).start()]

  else:

    return lease_value

In [46]:
# Applying regular expression function :-
data_clean['lease'] = data_clean['lease'].apply(clean_lease)

In [47]:
# unique values further more reduced to 137 from 223
len(data_clean['lease'].unique())

128

In [48]:
# checking unique values of the buudget :-
len(data_clean['budget'].unique())

1851

REDUCING THE UNIQUE VALUES OF THE BUDGET :-

In [49]:
def calculate_average(data):
    if re.match(r'£\d+ - £\d+ Per week', data):
        # extract the lower and upper bounds using string manipulation
        lower_bound = float(data.split('£')[1].split(' - ')[0])
        upper_bound = float(data.split('£')[2].split(' ')[0])
        # calculate the average
        average_price = str((lower_bound + upper_bound) / 2)
        return average_price
    elif re.match(r'£\d+ - £\d+ Per Week', data):
        # extract the lower and upper bounds using string manipulation
        lower_bound = float(data.split('£')[1].split(' - ')[0])
        upper_bound = float(data.split('£')[2].split(' ')[0])
        # calculate the average
        average_price = str((lower_bound + upper_bound) / 2)
        return average_price
    else:
        return data

In [50]:
# apply the function to the column and replace the original strings with the averages
data_clean['budget'] = data_clean['budget'].apply(calculate_average)

In [51]:
len(data_clean['budget'].unique())

1847

In [52]:
#  implementing one hot and label encoding :--
le = LabelEncoder()
label = data_clean['status']
label = le.fit_transform(label)
ohe = OneHotEncoder()
features = data_clean.drop(['status'],axis = 1)
features = ohe.fit_transform(features)
features = features.toarray()

In [53]:
# SCALING FEATURES :-
scaler = StandardScaler()
features = scaler.fit_transform(features)

In [54]:

# Splitting the data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(features , label, test_size=0.2, random_state=42)

In [55]:
# Training a Random Forest Regression model
rf = RandomForestRegressor(n_estimators=100, random_state=42)
rf.fit(X_train, y_train)


RandomForestRegressor(random_state=42)

In [56]:

# Predicting on the testing set
y_pred = rf.predict(X_test)

In [57]:
# converting y_pred into int
y_pred = y_pred.astype(int)

In [58]:
# computing accuracy score :-
acc = accuracy_score(y_test,y_pred)

In [59]:
print(acc)

0.9628670120898101


In [60]:
# computing precision score :-
prs = precision_score(y_test,y_pred)

print(prs)


0.9103448275862069


In [61]:
# full classification report of my dataset :-
print(classification_report(y_test,y_pred))

              precision    recall  f1-score   support

           0       0.96      1.00      0.98      8682
           1       0.91      0.45      0.61       582

    accuracy                           0.96      9264
   macro avg       0.94      0.73      0.79      9264
weighted avg       0.96      0.96      0.96      9264



In [63]:
# defining y_pred_class
y_p_c = ['LOST' if x > 0 else 'WON' for x in y_pred]


The accuracy and precision are given below, and for this, i used classfication report:
* ACCURACY SCORE :- 0.9628670120898101
* PRECISION SCORE :- 0.9103448275862069

In [65]:
print(y_p_c)

['WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'LOST', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'LOST', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'LOST', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'LOST', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'LOST', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'LOST', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON', 'WON',

we can conclude that our accuracy and precision is good on both training and testing data, so the model is pretty accurate.